pkg/render/intrusion_detection_test.go

```diff
@@ -207,18 +192,78 @@ var _ = Describe("Intrusion Detection rendering tests", func() {
+			{name: "policy.globalnetworkset", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "policy.serviceaccount", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "network.cloudapi", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "network.ssh", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "network.lateral.access", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "network.lateral.originate", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "dns.servfail", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "dns.dos", ns: "", group: "projectcalico.org", version: "v3", kind: "GlobalAlertTemplate"},
+			{name: "intrusion-detection", ns: "", group: "policy", version: "v1beta1", kind: "PodSecurityPolicy"},
+			{name: "intrusion-detection-psp", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
+			{name: "intrusion-detection-psp", ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
+			{name: render.DeepPacketInspectionNamespace, ns: "", group: "", version: "v1", kind: "Namespace"},
+			{name: render.NodeTLSSecretName, ns: render.DeepPacketInspectionNamespace, group: "", version: "v1", kind: "Secret"},
+			{name: render.TyphaTLSSecretName, ns: render.DeepPacketInspectionNamespace, group: "", version: "v1", kind: "Secret"},
+			{name: render.TyphaCAConfigMapName, ns: render.DeepPacketInspectionNamespace, group: "", version: "v1", kind: "ConfigMap"},
+			{name: render.DeepPacketInspectionName, ns: render.DeepPacketInspectionNamespace, group: "", version: "v1", kind: "ServiceAccount"},
+			{name: render.DeepPacketInspectionName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRole"},
+			{name: render.DeepPacketInspectionName, ns: "", group: "rbac.authorization.k8s.io", version: "v1", kind: "ClusterRoleBinding"},
+			{name: render.DeepPacketInspectionName, ns: render.DeepPacketInspectionNamespace, group: "apps", version: "v1", kind: "DaemonSet"},
+			{name: "tigera-secure", ns: "", group: "", version: "v1", kind: "IntrusionDetection"},
+		}
+
+		Expect(len(resources)).To(Equal(len(expectedResources)))
+
+		for i, expectedRes := range expectedResources {
+			rtest.ExpectResource(resources[i], expectedRes.name, expectedRes.ns, expectedRes.group, expectedRes.version, expectedRes.kind)
+		}
+
+		// IDS resource should be updated with default resource values for DPI
+		idsResp := rtest.GetResource(resources, "tigera-secure", "", "", "v1", "IntrusionDetection").(*operatorv1.IntrusionDetection)
+		Expect(len(idsResp.Spec.ComponentResources)).Should(Equal(1))
```

Comment:
We should add checks for clusterroles, clusterrolebindings, volumes, volume mounts, daemonset annotations.